### PR TITLE
Upgrade Log4j2 from 2.10.0 to 2.18.0

### LINKED ISSUE #2: Address CVE-2021-44228
We should address [CVE-2021-44228](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-44228)

[HIVE-25795](https://issues.apache.org/jira/browse/HIVE-25795) should help

### PR DESCRIPTION
**Pull request by mehdibn**
Fixes #2 

Applied Backports:
- https://github.com/apache/hive/commit/caf7ac0099645ac8500d824556941447e66e25e3
- https://github.com/apache/hive/commit/c9e7f5dd6191636232921279acc1a5dd5a6fcaff

### CODE REVIEW

File: testutils/ptest2/pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@ limitations under the License.
   <name>hive-ptest</name>
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <log4j2.version>2.10.0</log4j2.version>
+    <log4j2.version>2.12.1</log4j2.version>
```
Comment (rpignolet) [inline testutils/ptest2/pom.xml]: Why not `2.18.0` here ?

Comment (mehdibn) [reply testutils/ptest2/pom.xml:29]: never changed in apache hive : https://github.com/TOSIT-IO/hive/blob/44d6cf195568076956fe540cb2fe238831fe7124/testutils/ptest2/pom.xml#L29

Comment (mehdibn) [reply testutils/ptest2/pom.xml]: Actually, we don't build and deploy `ptest2` as it is a not included in the principal pom.xml as a module of the parent project.
Any way, i tested the build of this one with `2.18.0` and it succeeded. So, i will include it in the last commit.

**Review by rpignolet — APPROVED** [review]

**Comment by Pierrotws** [comment]
OK (cannot review)

**Pull request merged**